docs/pages/access-controls/sso/azuread.mdx

```diff
@@ -54,102 +60,116 @@ Before you get started you’ll need:
+8. Edit **Attributes & Claims**
 
-   ![Confirm Name Identifier](../../../img/azuread/azuread-8a-nameidentifier.png)
+   - Click on **Unique User Identifier (Name ID)** under **Required claim**.
+   - Change the "name identifier format" to **Default**. Make sure the source
+     attribute is `user.userprincipalname`.
 
-   - Add a group claim to make user security groups available to the connector
+     ![Confirm Name Identifier](../../../img/azuread/azuread-8a-nameidentifier.png)
 
-   ![Put in Security group claim](../../../img/azuread/azuread-8b-groupclaim.png)
+   - Add a group claim to make user security groups available to the connector:
 
-   - Add a claim that transforms an Azure AD username in order to pass it to Teleport.
+     ![Put in Security group claim](../../../img/azuread/azuread-8b-groupclaim.png)
 
-   ![Add a transformed username](../../../img/azuread/azuread-8c-usernameclaim.png)
+   - Add a claim that transforms an Azure AD username in order to pass it to
+     Teleport. Set the Source to "Transformation". In the new panel:
+      - Set the Transformation value to "Extract()"
+      - Set the Attribute name to `user.userprincipalname`.
+      - Set the Value to `ToLowercase()`.
 
-9. In **SAML Signing Certificate**, click the link to download the **Federation Metadata XML**.
+     ![Add a transformed username](../../../img/azuread/azuread-8c-usernameclaim.png)
 
-   ![Download Federation Metadata XML](../../../img/azuread/azuread-9-fedmeatadataxml.png)
+9. In **SAML Certificates** section, copy the **App Federation
+   Metadata URL** link and save it for the next step:
 
-<Admonition
-  type="warning"
-  title="Important"
->
-  This is an important document.  The Federation Metadata XML file sets the SSO location and other important values. Transmit the Federation Metadata XML file securely so it is not malformed.  Using the `entity_descriptor_url` setting allows for loading from the source.
-</Admonition>
+   ![Download Federation Metadata XML](../../../img/azuread/azuread-9-fedmeatadataxml.png)
 
 ## Create a SAML Connector
 
-Now, create a SAML connector resource. Write the following to `azure-connector.yaml`:
+Now, create a SAML connector resource using `tctl`.
+
+```code
+$ tctl sso configure saml --preset ad \
+--entity-descriptor https://login.microsoftonline.com/ff882432.../federationmetadata/2007-06/federationmetadata.xml\?appid\=b8d06e01... \
+--attributes-to-roles http://schemas.microsoft.com/ws/2008/06/identity/claims/groups,41c94563...,dev \
+--attributes-to-roles http://schemas.microsoft.com/ws/2008/06/identity/claims/groups,8adac502...,access > azure-connector.yaml
+```
+
+In the example above:
+
+- `--entity-descriptor` is give the app federation metadata URL saved in the
+  previous step.
+- Each `--attributes-to-roles` is give the name of the schema definition for
+  groups, the UUID of an AD group, and the Teleport role members of that group
+  will be assigned.
```

Comment:
```suggestion
  will be assigned to.
```

Reply:
Gonna push back on this on a bit. If it were a "group", then the user would be assigned "to" it. But a role, that is something that is assigned _to_ the user.